enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/vault/SecretNode.java

```diff
@@ -41,13 +56,81 @@ public SecretNode(SecretItem vault) {
         setName(vault.getName());
         setDisplayName(vault.getName());
         setIconBaseWithExtension(SECRET_ICON);
-        setShortDescription(vault.getDescription());
+        setShortDescription(
+                createShortDescription(
+                        vault.getLifecycleState(),
+                        vault.getDeletionTime()));
+    }
+    
+    private String createShortDescription(String state, Date deletionTime) {
+        if (deletionTime != null) {
+            return Bundle.SecretNodeDesc(this.getItem().getName(), state, getDeletionTimeInfo(deletionTime));
+        }
+        return Bundle.SecretNodeDesc(this.getItem().getName(), state, "");
+    }
+    
+    private String getDeletionTimeInfo(Date deletionTime) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        return "\nDeletion time: " + dateFormat.format(deletionTime);
     }
 
     public static NodeProvider<SecretItem> createNode() {
         return SecretNode::new;
     }
+    
+    @Override
+    public void update(OCIItem item) {
```

Comment:
Make sure `update` does not run in a LSP messaging thread, or other 'must not block' thread ...